VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/AsToCast/AsToCastCodeFix.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace VSDiagnostics.Diagnostics.General.AsToCast
+{
+    [ExportCodeFixProvider("SingleEmptyConstructorCodeFix", LanguageNames.CSharp), Shared]
+    public class AsToCastCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(AsToCastAnalyzer.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var statement = root.FindNode(diagnosticSpan);
+            context.RegisterCodeFix(CodeAction.Create("Use cast instead of as", x => RemoveConstructorAsync(context.Document, root, statement), nameof(AsToCastAnalyzer)), diagnostic);
+        }
+
+        private Task<Solution> RemoveConstructorAsync(Document document, SyntaxNode root, SyntaxNode statement)
+        {
+            var binaryExpression = (BinaryExpressionSyntax) statement;
+            var typeSyntax = SyntaxFactory.ParseTypeName(binaryExpression.Right.GetText().ToString());
```

Comment:
I don't like this, it feels rather error-prone. Can you add the following test case and see if it breaks anything? (Even if it doesn't, add it anyway):

    void Main()
    {
	    bool? b = GetBoxedType() as bool?;
    }

    object GetBoxedType()
    {
	    return true;
    }